src/pudl/helpers.py

```diff
@@ -2185,3 +2185,305 @@ def retry(
             )
             time.sleep(delay)
     return func(**kwargs)
+
+
+def standardize_phone_column(df: pd.DataFrame, columns: list[str]) -> pd.DataFrame:
+    """Standardize phone numbers in the specified columns of the DataFrame.
+    US numbers: ###-###-####
+    International numbers with the international code at the beginning.
+    Numbers with extensions will be appended with "x#".
+    Non-numeric entries will be returned as np.nan. Entries with fewer than
+    10 digits will be returned with no hyphens.
+
+    Args:
+    df: The DataFrame to modify.
+    columns: A list of the names of the columns that need to be standardized
+    Returns:
+    The modified DataFrame with standardized phone numbers in the same column.
+    """
+
+    # Function to clean and standardize phone numbers
+    def standardize_phone_number(phone: str) -> str:
+        # Check if the value is NaN, return it as is
+        if pd.isna(phone):
+            return phone
+
+        # Split phone number and extension, if present
+        phone_parts = re.split(r"[xX]", str(phone))
+        phone_main = phone_parts[0]  # Main phone number
+        extension = (
+            phone_parts[1].strip() if len(phone_parts) > 1 else None
+        )  # Extension, if it exists
+
+        # Remove unwanted characters (parentheses, spaces, periods, and dashes) from the main phone number
+        phone_main = re.sub(
+            r"[^\d]", "", phone_main.replace(".0", "")
```

Comment:
There were lots of phone numbers in the raw data with ".0" at the end of the value. So I'm using `replace` to remove those first.

Reply:
You might want to use a regex that only matches this pattern at the end of the string to avoid accidentally removing characters from the middle of a number that could be reported like 303.443.0123

Reply:
Good point. I'm removing that logic. The regex should be all that's needed here.

Reply:
Actually I'm gonna leave in one line toward the top to get rid of these substrings just to make sure the rest of the number standardization is working with cleaner data.

Updated.